src/tool_cb_hdr.c

```diff
@@ -327,3 +321,23 @@ static char *parse_filename(const char *ptr, size_t len)
 
   return copy;
 }
+
+static bool is_etag_btw_double_quotes(char *ptr)
+{
+  int i = 0, count = 0;
+  const char double_quote = '\"';
+  const unsigned char max_double_quotes = 2;
+
+  while(ptr[i] != '\0') {
+    if(ptr[i] == double_quote) {
+      count++;
+    }
+    i++;
+  }
+
+  if(count != max_double_quotes) {
+    return FALSE;
+  }
```

Comment:
An optimization here would be to exit the loop if a third quote is found as then you know its wrong and there's no need to continue...

Reply:
Oh, and this function will also be fine with `Etag: yadayada "23232"` which seems wrong.

Reply:
> An optimization here would be to exit the loop if a third quote is found as then you know its wrong and there's no need to continue...

Agree


> Oh, and this function will also be fine with `Etag: yadayada "23232"` which seems wrong.

Indeed :/, from [mozilla](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) documentation we have:
```
ETag: "33a64df551425fcc55e4d42a148795d9f25f89d4"
ETag: W/"0815"
```
@bagder Would it be better to check firstly, if the first character is equal to `"` or the 3 starting characters are `\W"`?

Reply:
The `W/` prefix is for "weak" etag ([explained here](https://tools.ietf.org/html/rfc7232#section-2.3)) and we haven't supported that before so I think you can skip those now as well.

Reply:
Because header value could have any number of LWS in prefix we have to take care of trimming them before checking if the first character is a double quote. My last commit handle this behavior 01866702b9ed6a90be23fd5cf64d02220db5f005